### PR TITLE
MM-1214: DefaultTableAliasedSchema.equals for schema comparison.

### PR DESCRIPTION
**Pull request by jhorcicka**
Fixes MM-1214. 

DefaultTableAliasedSchema.equals(...) method was added to use "wrappedSchema" field for comparison.

### CODE REVIEW

**Comment by kaspersorensen** [comment]
I wonder if this is sensible. Would you really say that the "alias" schema is equal to the schema that it's aliasing? I think that may create other problems down the line.

If it turns out that I'm wrong then I have another remark, namely that the `hashCode()` method should be implemented according to the same logic then.

**Comment by jhorcicka** [comment]
Good point. 

I tried another approach instead. In `CompositeDataContext.getDataContext` I can check the actual type of `dcSchema` and cast it. I believe it's less intrusive now. What is your opinion about that, @kaspersorensen?

File: core/src/main/java/org/apache/metamodel/CompositeDataContext.java [hunk 1]
```diff
@@ -122,7 +123,8 @@ private DataContext getDataContext(Table table) {
                         Schema dcSchema = dc.getSchemaByName(schema.getName());
                         if (dcSchema != null) {
                             // second round = try with schema equals method
-                            if (dcSchema.equals(schema)) {
+                            if (dcSchema.equals(schema) || (dcSchema instanceof DefaultTableAliasedSchema
```
Comment (kaspersorensen) [inline core/src/main/java/org/apache/metamodel/CompositeDataContext.java]: I would use the interface `WrappingSchema` here.

Comment (jhorcicka) [reply core/src/main/java/org/apache/metamodel/CompositeDataContext.java]: Updated. Thanks for the tip.

**Comment by kaspersorensen** [comment]
And now you unfortunately fell victim to one of our sporadically failing tests. But LGTM now.

**Comment by kaspersorensen** [comment]
Please also update the CHANGES.md file. Then I will merge.

**Comment by jhorcicka** [comment]
Updated.

File: CHANGES.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ### Apache MetaModel _WIP_
 
- * [METAMODEL-1151] - Created DataContextFactory implementations for each remaining module.
```
Comment (kaspersorensen) [inline CHANGES.md]: We shouldn't remove this item

Comment (jhorcicka) [reply CHANGES.md]: Oh, sure. That was my merging error. Updated.

**Pull request merged**